src/lib/wifi-control-request.js

```diff
@@ -0,0 +1,251 @@
+const fs = require('fs-extra');
+const JOIN_NETWORK_TIMEOUT = 30000;
+
+module.exports = class WiFiControlRequest {
+	constructor(deviceId, { ui, newSpin, stopSpin, file }) {
+		this.deviceId = deviceId;
+		this.device = null;
+		this.ui = ui;
+		this.newSpin = newSpin;
+		this.stopSpin = stopSpin;
+		this.file = file;
+	}
+
+	async configureWifi() {
+		// prompt for scan networks
+		let network;
+		if (this.file) {
+			network = await this.getNetworkToConnectFromJson();
+		} else {
+			network = await this.getNetworkToConnect();
+		}
+
+		if (network) {
+			const success = await this.joinWifi(network);
+			if (success) {
+				this.ui.stdout.write('Successfully connected to Wi-Fi');
+				this.ui.stdout.write(`${os.EOL}`);
+				this.ui.stdout.write(`Done! Your device should now restart.${os.EOL}`);
+				process.exit(0);
+			} else {
+				this.ui.stderr.write('Failed to connect to Wi-Fi');
+				this.ui.stderr.write(`${os.EOL}`);
+				process.exit(1);
+			}
+		} else {
+			this.ui.stdout.write(`${os.EOL}`);
+			this.ui.stdout.write('No network selected');
+			this.ui.stdout.write(`${os.EOL}`);
+			process.exit(1);
+		}
+	}
+
+	async getNetworkToConnectFromJson() {
+		try {
+			const { network, password } = await fs.readJSON(this.file);
+			if (!network) {
+				this.ui.stderr.write('No SSID found in the file');
+				return null;
+			}
+			return { ssid: network, password };
+		} catch (error) {
+			throw new Error('Ups! We could not read the file. Please try again: ' + error.message);
+		}
+
+	}
+
+	async getNetworkToConnect({ prompt = true } = { }) {
+		try {
+			let scan = true;
+			if (prompt) {
+				scan = await this.promptForScanNetworks();
+			}
+			if (scan) {
+				const networks = await this.scanNetworks();
+				if (networks.length) {
+					const network = await this.promptToSelectNetwork(networks);
+					if (network?.rescan){
+						return await this.getNetworkToConnect({ prompt: false });
+					} else {
+						return network;
+					}
+				}
+			}
+			return await this.pickNetworkManually();
+		} catch (error) {
+			throw new Error('Ups! something went wrong. Please try again.', error.message);
+		}
+	}
+
+	async promptForScanNetworks() {
+		const question = {
+			type: 'confirm',
+			name: 'scan',
+			message: 'Would you like to scan for Wi-Fi networks?'
+		};
+		const ans = await this.ui.prompt([question]);
+		return ans.scan;
+	}
+
+	async scanNetworks() {
+		// open device by id
+		this.newSpin('Scanning for Wi-Fi networks').start();
+		let networks = [];
+		networks = await this._deviceScanNetworks();
+		this.stopSpin();
+		if (!networks.length) {
+			const answers = await this.ui.prompt([{
+				type: 'confirm',
+				name: 'rescan',
+				message: 'Uh oh, no networks found. Try again?',
+				default: true
+			}]);
+			if (answers.rescan){
+				return this.scanNetworks();
+			}
+		}
+		return this._filterNetworks(networks);
+	}
+
+	_filterNetworks(networkList) {
+		return networkList.filter((ap) => {
```

Comment:
Also filter duplicate ssids. I have 3 access points for my Wi-Fi network across the house so I see this:
```
? Select the Wi-Fi network with which you wish to connect your device: (Use arrow keys)
  ──────────────
❯ [rescan networks] 
  ──────────────
  bunker 
  bunker 
  bunker 
  SONOS-AyMBBwACH1QqG9wCrgM= 
(Move up and down to reveal more choices)
```